xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/internal/rss/AbstractNotificationRSSRenderer.java

```diff
@@ -0,0 +1,122 @@
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.notifications.internal.rss;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.localization.ContextualLocalizationManager;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.notifications.CompositeEvent;
+import org.xwiki.notifications.NotificationException;
+import org.xwiki.notifications.internal.ModelBridge;
+import org.xwiki.notifications.rss.NotificationRSSRenderer;
+
+import com.rometools.rome.feed.synd.SyndEntry;
+import com.rometools.rome.feed.synd.SyndFeed;
+import com.rometools.rome.feed.synd.SyndFeedImpl;
+
+/**
+ * Abstract definition of the {@link NotificationRSSRenderer} API.
+ *
+ * @version $Id$
+ * @since 9.6RC1
+ */
+public abstract class AbstractNotificationRSSRenderer implements NotificationRSSRenderer
+{
+    @Inject
+    protected ContextualLocalizationManager contextualLocalizationManager;
+
+    @Inject
+    protected ModelBridge modelBridge;
+
+    @Inject
+    protected DocumentReferenceResolver<String> documentReferenceResolver;
+
+    @Inject
+    private ComponentManager componentManager;
+
+    @Inject
+    protected Logger logger;
+
+    @Override
+    public SyndFeed renderFeed(List<CompositeEvent> events)
+    {
+        SyndFeed feed = new SyndFeedImpl();
+
+        // Define the general properties of the rss
+        feed.setFeedType("rss_2.0");
+        feed.setTitle(this.contextualLocalizationManager.getTranslationPlain("notifications.rss.feedTitle"));
+
+        // Set the RSS feed link to the service generating the feed
+        feed.setLink(this.modelBridge.getDocumentURL(
+                this.documentReferenceResolver.resolve("XWiki.Notifications.Code.NotificationRSSService"),
+                "get", "outputSyntax=plain"));
+
+        // Set the feed description
+        feed.setDescription(this.contextualLocalizationManager.getTranslationPlain(
+                "notifications.rss.feedDescription"));
+
+        // Add every given CompositeEvent entry to the rss
+        List<SyndEntry> entries = new ArrayList<>();
+        for (CompositeEvent event : events) {
+            try {
+                NotificationRSSRenderer renderer = this.getRenderer(event);
```

Comment:
This works indeed. But @tmortagne had suggested something a bit different that I have implemented for the Notification Email. It's the default renderer itself that look if a more specific renderer exists. See: https://github.com/xwiki/xwiki-platform/blob/5248ea6d230eba01ed012e2a5419381c6074467c/xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/internal/email/DefaultNotificationEmailRenderer.java#L58-L58

However what your propose works too and it's the way I have implemented the Notification Renderer.